src/middleware.rs

```diff
@@ -1,10 +1,9 @@
+use super::Result;
 use crate::{Counter, Job, RedisPool, UnitOfWork, WorkerRef};
 use async_trait::async_trait;
-use slog::error;
 use std::sync::Arc;
 use tokio::sync::RwLock;
-
-pub type ServerResult = Result<(), Box<dyn std::error::Error>>;
```

Comment:
Makes sense to me. I'd prefer a root level error with types than an alias with semi-fixed type params. Since we're already moving a few things around, I think it's time. 👍 .